packages/web/src/redux/song/song-actions.js

```diff
@@ -1,90 +1,157 @@
 import api from "../../api";
-import * as songTypes from "./song-type";
+import { normalizeSongs } from "../../schema/song-schema";
 import * as auth from "../../services/auth";
+import * as SongTypes from "./song-type";
+import { songsTypes } from "./song-type";
 
-export const fetchSongRequest = () => {
-  return { type: songTypes.FETCH_SONG_REQUEST };
+export const fetchSongsRequest = () => {
+  return { type: SongTypes.FETCH_SONG_REQUEST };
 };
 
-export const fetchSongError = (message) => {
-  return { type: songTypes.FETCH_SONG_ERROR, payload: message };
+export const fetchSongsError = (message) => {
+  return { type: SongTypes.FETCH_SONG_ERROR, payload: message };
 };
 
-export const fetchSongSuccess = (data) => {
-  return { type: songTypes.FETCH_SONG_SUCCESS, payload: data };
+export const fetchSongsSuccess = ({
+  type = songsTypes.ALL_SONGS,
+  songsByID,
+  songsIds,
+}) => {
+  return {
+    type: SongTypes.FETCH_SONG_SUCCESS,
+    payload: { type: type, songsByID: songsByID, songsIds: songsIds },
+  };
 };
 
-export const fetchSongReset = () => {
-  return { type: songTypes.FETCH_SONG_RESET };
+export function fetchSongs(fetchType) {
+  switch (fetchType) {
+    case songsTypes.ALL_SONGS:
+      return fetchAllSongs();
+    case songsTypes.MY_SONGS:
+      return fetchMySongs();
+    case songsTypes.FAVORITE:
+      return fetchFavoriteSongs();
+    default:
+      break;
+  }
+  return fetchAllSongs();
+}
+
+export const fetchSongsReset = () => {
+  return { type: SongTypes.FETCH_SONG_RESET };
 };
 
-export const fetchSong = () => {
+// Fetch all songs
+
+export const fetchAllSongs = () => {
   return async function fetchSongThunk(dispatch) {
-    dispatch(fetchSongRequest());
+    dispatch(fetchSongsRequest());
 
     try {
       const songs = await api.getAllSongs();
 
       if (songs.errorMessage) {
-        return dispatch(fetchSongError(songs.errorMessage));
+        return dispatch(fetchSongsError(songs.errorMessage));
+      }
+
+      const normalizedSongs = normalizeSongs(songs.data);
+
+      return dispatch(
+        fetchSongsSuccess({
+          type: songsTypes.ALL_SONGS,
+          songsByID: normalizedSongs.entities.songs,
+          songsIds: normalizedSongs.result,
+        }),
+      );
+    } catch (error) {
+      return dispatch(fetchSongsError(error.message));
+    }
+  };
+};
+
+export const fetchMySongs = () => {
+  return async function fetchMySongThunk(dispatch) {
+    dispatch(fetchSongsRequest());
+
+    const token = await auth.getCurrentUserToken();
+
+    if (!token) {
+      return dispatch(fetchSongsError("User token null"));
+    }
+
+    try {
+      const MySongs = await api.getMeSongs({
+        Authorization: `Bearer ${token}`,
+      });
```

Comment:
```suggestion
      const mySongs = await api.getMeSongs({
        Authorization: `Bearer ${token}`,
      });
```

By convention, variables with Upper CamelCase are classes.